go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -72,9 +68,6 @@ func (npw *nodePortWatcher) AddService(service *kapi.Service) {
 			// Table 2 handles return NAT traffic for node port and forwards out of the host
 			// NodePort/Ingress access in the OVS bridge will only ever come from outside of the host
 			for _, ing := range service.Status.LoadBalancer.Ingress {
-				if ing.IP == "" {
-					continue
-				}
```

Comment:
This isn't redundant validation; there are two different ways to fill in `Status.LoadBalancer.Ingress[*]`; you can set `IP` (GCP/Azure) or you can set `HostName` (AWS). We only do the extra processing here for load balancers that do it the first way.